<launch>

  <!-- Arguments -->
  <arg name="file" doc="Specify the bag file name"/>
  <arg name="car_name" default="$(env CAR_NAME)" doc="Set the CAR_NAME environment variable to 'anicar2' or 'anicar3' prior to starting this launch file" />
  
  <!-- Parameters -->
  <param name="/use_sim_time" value="true" />
   
  <!-- Play rosbag -->
  <node pkg="rosbag" type="play" name="player" output="screen" args="-l --clock $(arg file)"/>

  <!-- Processing pipeline -->
  <arg name="scale" default="0.5" />
  <node name="resized" type="resize" pkg="image_preproc_ros_tool" ns="/sensors/camera/left">
    <remap from="/sensors/camera/left/src/image_camera/image" to="/sensors/camera/left/image_raw" />
    <remap from="/sensors/camera/left/src/src/image_camera/camera_info" to="/sensors/camera/left/camera_info" />
    <remap from="/sensors/camera/left/tgt/image_camera/image" to="/sensors/camera/left/resized/image_raw"/>
    <remap from="/sensors/camera/left/tgt/image_camera/camera_info" to="/sensors/camera/left/resized/camera_info" />
    <param name="scale" type="double" value="$(arg scale)" />
  </node>
  <node name="resized" type="resize" pkg="image_preproc_ros_tool" ns="/sensors/camera/right">
    <remap from="/sensors/camera/right/src/image_camera/image" to="/sensors/camera/right/image_raw" />
    <remap from="/sensors/camera/right/src/image_camera/camera_info" to="/sensors/camera/right/camera_info" />
    <remap from="/sensors/camera/right/tgt/image_camera/image" to="/sensors/camera/right/resized/image_raw"/>
    <remap from="/sensors/camera/right/tgt/image_camera/camera_info" to="/sensors/camera/right/resized/camera_info" />
    <param name="scale" type="double" value="$(arg scale)" />
  </node>
  <include file="$(find vehicle_setup)/launch/urdf_publisher.launch" />

  <!-- Pylon detector -->
  <node pkg="demo_image_processing_ros_tool" type="pylon_detector" name="pylon_detector" output="screen">
    <rosparam command="load" file="$(find demo_image_processing_ros_tool)/launch/params/pylon_detector.yaml"/>
  </node>
  
  <!-- Visualization -->
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find demo_image_processing_ros_tool)/res/pylon_detector.rviz" />
  
  <!-- Reconfiguration -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure"/>

</launch>
